src/cli/pm_version_command.zig

```diff
@@ -0,0 +1,605 @@
+    const VersionType = enum {
+        patch,
+        minor,
+        major,
+        prepatch,
+        preminor,
+        premajor,
+        prerelease,
+        specific,
+        from_git,
+
+        pub fn fromString(str: []const u8) ?VersionType {
+            if (strings.eqlComptime(str, "patch")) return .patch;
+            if (strings.eqlComptime(str, "minor")) return .minor;
+            if (strings.eqlComptime(str, "major")) return .major;
+            if (strings.eqlComptime(str, "prepatch")) return .prepatch;
+            if (strings.eqlComptime(str, "preminor")) return .preminor;
+            if (strings.eqlComptime(str, "premajor")) return .premajor;
+            if (strings.eqlComptime(str, "prerelease")) return .prerelease;
+            if (strings.eqlComptime(str, "from-git")) return .from_git;
+            return null;
+        }
+    };
+
+    pub fn exec(ctx: Command.Context, pm: *PackageManager, positionals: []const string, original_cwd: []const u8) !void {
+        const package_json_dir = findPackageDir(ctx.allocator, original_cwd) catch original_cwd;
+
+        if (positionals.len <= 1) {
+            try showHelp(ctx, pm, package_json_dir);
+            return;
+        }
+
+        const version_type, const new_version = parseVersionArgument(positionals[1]);
+
+        try verifyGit(package_json_dir, pm);
+
+        var path_buf: bun.PathBuffer = undefined;
+        const package_json_path = bun.path.joinAbsStringBufZ(package_json_dir, &path_buf, &.{"package.json"}, .auto);
+
+        const package_json_contents = bun.sys.File.readFrom(bun.FD.cwd(), package_json_path, ctx.allocator).unwrap() catch |err| {
+            Output.errGeneric("Failed to read package.json: {s}", .{@errorName(err)});
+            Global.exit(1);
+        };
+        defer ctx.allocator.free(package_json_contents);
+
+        const package_json_source = logger.Source.initPathString(package_json_path, package_json_contents);
+        const json = JSON.parsePackageJSONUTF8(&package_json_source, ctx.log, ctx.allocator) catch |err| {
+            Output.errGeneric("Failed to parse package.json: {s}", .{@errorName(err)});
+            Global.exit(1);
+        };
+
+        const scripts = json.asProperty("scripts");
+        const scripts_obj = if (scripts) |s| if (s.expr.data == .e_object) s.expr else null else null;
+
+        if (scripts_obj) |s| {
+            if (s.get("preversion")) |script| {
+                if (script.asString(ctx.allocator)) |script_command| {
+                    try RunCommand.runPackageScriptForeground(
+                        ctx,
+                        ctx.allocator,
+                        script_command,
+                        "preversion",
+                        package_json_dir,
+                        pm.env,
+                        &.{},
+                        pm.options.log_level == .silent,
+                        ctx.debug.use_system_shell,
+                    );
```

Comment:
```suggestion
        if (pm.options.do.run_scripts) {
            if (scripts_obj) |s| {
```

scripts should not run if --ignore-scripts is passed